### PR TITLE
Python 3.8 build env

### LINKED ISSUE #453: Add Python3.8 CI build
Python 3.8 has been out since Oct '19. Add it as a build environment. Potentially might just ignore if it fails.

### PR DESCRIPTION
**Pull request by Amertz08**
#### Title

Please **pick a concise, informative and complete title** for your PR.
The title is important because it will appear in [our change log](https://github.com/RaRe-Technologies/smart_open/blob/master/CHANGELOG.md).

#### Motivation

Please explain the motivation behind this PR in the description.

If you're fixing a bug, link to the issue number like so:

- Fixes #453 

If you're adding a new feature, then consider opening a ticket and discussing it with the maintainers before you actually do the hard work.

#### Tests

If you're fixing a bug, consider [test-driven development](https://en.wikipedia.org/wiki/Test-driven_development):

1. Create a unit test that demonstrates the bug. The test should **fail**.
2. Implement your bug fix.
3. The test you created should now **pass**.

If you're implementing a new feature, include unit tests for it.

Make sure all existing unit tests pass.
You can run them locally using:

    pytest smart_open

If there are any failures, please fix them before creating the PR (or mark it as WIP, see below).

#### Work in progress

If you're still working on your PR, include "WIP" in the title.
We'll skip reviewing it for the time being.
Once you're ready to review, remove the "WIP" from the title, and ping one of the maintainers (e.g. mpenkov).

#### Checklist

Before you create the PR, please make sure you have:

- [ ] Picked a concise, informative and complete title
- [ ] Clearly explained the motivation behind the PR
- [x] Linked to any existing issues that your PR will be solving
- [ ] Included tests for any new functionality
- [ ] Checked that all unit tests pass

#### Workflow

Please avoid rebasing and force-pushing to the branch of the PR once a review is in progress.
Rebasing can make your commits look a bit cleaner, but it also makes life more difficult from the reviewer, because they are no longer able to distinguish between code that has already been reviewed, and unreviewed code.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -25,6 +25,13 @@ jobs:
         - BOTO_CONFIG: "/dev/null"
         - SO_ENABLE_MOTO_SERVER: "1"
         - TOXENV: "check_keys,py37-doctest,enable_moto_server,py37-test,py37-benchmark,py37-integration,disable_moto_server"
+    - python: '3.8'
```
Comment (mpenkov) [inline .travis.yml:28]: Please disable Py3.8 benchmarking for now.

I'm in the process of rewriting the benchmarks so we won't need a separate bucket for each Python version.

We also probably don't want to bother with the Moto server for all builds.

Comment (Amertz08) [reply .travis.yml:28]: So drop `py38-benchmark` and both the moto builds as well as the SO_ENABLE_MOTO_SERVER?

Comment (mpenkov) [reply .travis.yml:28]: Yes

Comment (Amertz08) [reply .travis.yml:28]: Got it. Thank you for the clarification.

**Pull request merged**